features/2025-07-10-offline-install-feature-spec.md

```diff
@@ -0,0 +1,386 @@
+
+The `--skip-contour-install` does not install Contour. Therefore the Gateway resource types will not function until extensibility is implemented.
+
+> [!CAUTION]
+>
+> `--skip-contour-install` is expected to be removed from Radius in the near future as part of the compute extensibility work. The design for the Gateway recipe no longer uses Contour and Contour is no longer installed by default.
+
+If a Helm mirror has not been configured via `helm repo add`, the `--chart` argument can be used.
+
+```bash
+rad install kubernetes \
+  --chart <PATH_TO_RADIUS_HELM_CHART_FROM_HELM_PULL> \
+  ...
+```
+
+### User Story 4 – Install Radius via Helm
+
+***As a platform engineer, I need to install Radius using Helm***
+
+The platform engineer installs Radius using Helm with similar settings.
+
+```bash
+helm upgrade radius radius/radius \
+  --install \
+  --create-namespace 
+  --namespace radius-system \
+  --version 0.48.0 \
+  --wait --timeout 15m0s \
+  --set image.repository=<OCI_REGISTRY_URL>/<REPOSITORY>
+```
+
+## Scenario 2 – Managing Terraform
+
+### User Story 5 – Install Terraform
+
+***As a platform engineer, I need to install and configure Terraform for Radius to use***
+
+The platform engineer deploys a Terraform resource. The Terraform resource replaced the `recipeConfig` properties in the Environment today and adds all other [terraformrc configuration options](https://developer.hashicorp.com/terraform/cli/config/config-file).
+
+**`terraform.bicep`**
+
+```yaml
+param token string
+
+resource tf 'System.Resources/terraform@2025-08-01-preview' = {
+  name: 'tf'
+  terraformCLI: {
+    url: 'https://<TF_MIRROR_URL>/terraform_1.5.7_linux_amd64.zip'
+    checksum: 'sha256:37f2d497ea512324d59b50ebf1b58a6fcc2a2828d638a4f6fdb1f41af00140f3'
+  }
+  terraformrc: {
+    provider_installation: {
+      network_mirror: {
+        path: 'https://<TF_MIRROR_URL>/'
+        include: ["*"]
+      }
+      direct: {
+        exclude: ["azurerm"]
+      }
+    }
+    // Equivilent to recipeConfig.authentication on Environments today
+    credentials: {
+      <TF_MIRROR_URL>: {
+        secret: providerSecret.id
+      }
+    }
+    // Environment variables injected into the Terraform process
+    // Equivilent to recipeConfig.env on Environments today
+    env: {
+      TF_REGISTRY_CLIENT_TIMEOUT: 15
+    }
+  }
+}
+
+resource providerSecret 'Applications.Core/secretStores@2023-10-01-preview' = {
+  name: 'providerSecret'
+  properties: {
+    type: 'generic'
+    data: {
+      token: {
+        value: token
+      }
+    }
+  }
+}
+```
+
+The Terraform resource intentionally does not include these terraformrc properties:
+
+* `credentials_helper`
+* `provider_installation.filesystem_mirror`
+* `disable_checkpoint`
+* `disable_checkpoint_signature`
+* `plugin_cache_dir`
+
+The platform engineer deploys the Terraform resource.
+
+````bash
+$ rad deploy terraform.bicep
```

Comment:
I realize that this is using the existing Radius deploy mechanism to deploy an instance of Terraform, which makes sense, but let's discuss if this could be confusing because it's like a special kind of `rad deploy` which Radius is able to find and integrate with.